### ISSUE TITLE
jsx-a11y/img-redundant-alt issue about image word is sub of long word

### ISSUE THREAD

**Issue by markEHVN** [jsx-eslint/eslint-plugin-jsx-a11y#1037]
### Description
This rule can not caught `image` word inside a word.

```
export const FormHeader = () => (
  <div>
    <img src="./my-family.jqp" alt="familyImage" />
  </div>
);
```

I try oxlint, the tools can caught the above problem. I hope we should implement it. 

You can see source code [at here](https://github.com/markEHVN/compare-eslint-oxlint-plugin/tree/jsx-a11y/img-redundant-alt).

**Comment by markEHVN** [comment]
I think `familyImage` only is an example. We can change sentence to `This is an familyimage`, I think we should caught an error at here.

**Comment by ljharb** [comment]
right, but "family image" is what someone would type. just because a word has the letters "image" in it doesn't mean we should warn on it - "pilgrimage" is one of many examples that are valid.

**Comment by markEHVN** [comment]
Yep, I understood. 

> Enforce img alt attribute does not contain the word image, picture, or photo.

I read above definition so I think we also should prevent user type "something image" or "somethingimage". "pilgrimage" may be is a special word.

**Comment by ljharb** [comment]
The former yes, but the latter, no. If someone types incorrect english, we can't really do anything about that.

**Comment by markEHVN** [comment]
Yep, thanks. I think we should create [a new succeed example for this rule](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/img-redundant-alt.md#succeed), it will help people have more context.

**Comment by ljharb** [comment]
I don't understand why anyone would ever type a camelCase word into an alt attribute in the first place, though.

**Comment by ljharb** [comment]
What I'm suggesting is that it's not possible to help this kind of mistake without causing worse problems, and so it might just have to be relegated to human review.

**Comment by markEHVN** [comment]
Agreed. Thanks for your attention 👍
